src/collectors/mesos/mesos.py

```diff
@@ -12,55 +12,187 @@
+            r[executor_id]['framework_id'] = i['framework_id']
+            r[executor_id]['statistics'] = r[executor_id].get('statistics', {})
+            executor = r.get(executor_id, {'statistics': {}})
+            grouped_statistics = executor['statistics']
+            r[executor_id]['statistics'] = self._sum_statistics(
+                i['statistics'], grouped_statistics
+            )
+        return r
+
+    def _sum_statistics(self, x, y):
+        stats = set(x) | set(y)
+        summed_stats = {}
+        for k in stats:
+            summed_stats.update({k: x.get(k, 0) + y.get(k, 0)})
+        return summed_stats
+
+    def _collect_slave_statistics(self):
+        result = self._get(
+            self.config['host'],
+            self.config['port'],
+            'monitor/statistics.json'
+        )
+        if not result:
+            return
+
+        result = self._group_tasks_statistics(result)
+
+        for executor_id, executor in result.iteritems():
+            executor_statistics = executor['statistics']
+            for key in executor_statistics:
+                value = executor_statistics[key]
+                framework_id = self.known_frameworks[executor['framework_id']]
+                framework = self._sanitize_metric_name(framework_id)
+                executor_name = self._sanitize_metric_name(executor_id)
+                metric = 'frameworks.%s.executors.%s.%s' % \
+                         (framework, executor_name, key)
+                self.publish(metric, value, precision=self._precision(value))
+
+    def _get(self, host, port, path):
+        """
+        Execute a Mesos API call.
+        """
+        url = 'http://%s:%s/%s' % (host, port, path)
         try:
-            url = "http://%s:%s/%s" % (self.config['host'],
-                                       self.config['port'],
-                                       self.METRICS_PATH)
+            response = urllib2.urlopen(url)
+        except Exception, err:
+            self.log.error("%s: %s", url, err)
+            return False
 
-            return json.load(urllib2.urlopen(url))
-        except (urllib2.HTTPError, ValueError), err:
-            self.log.error('Unable to read JSON response: %s' % err)
-            return {}
-
-    def clean_up(self, text):
-        return text.replace('/', '.')
+        try:
+            doc = json.load(response)
+        except (TypeError, ValueError):
+            self.log.error("Unable to parse response from Mesos"
+                           " as a json object")
+            return False
+
+        return doc
+
+    def _precision(self, value):
+        """
+        Return the precision of the number
+        """
+        value = str(value)
+        decimal = value.rfind('.')
+        if decimal == -1:
+            return 0
+        return len(value) - decimal - 1
+
+    def _sanitize_metric_name(self, name):
+        return name.replace('.', '_').replace('/', '_')
```

Comment:
It's used only on `framework_id` and `executor_id`. So for this input

```json
...
  "frameworks": [
    {
      "checkpoint": true,
      "completed_executors": [],
      "executors": [],
      "failover_timeout": 604800,
      "hostname": "master.mesos.local",
      "id": "105146",
      "name": "marathon-0.7.6",
      "role": "*",
      "user": "root"
    }
  ]
...
```
```json
{
    "executor_id": "com.domain.group_anotherApp.08002432371b",
    "executor_name": "Command Executor (Task: com.domain.group_anotherApp.08002432371b) (Command: sh -c 'start...')",
    "framework_id": "105146",
    "source": "com.domain.group_anotherApp.08002432371b",
    "statistics": {
      "cpus_limit": 1.6,
      "cpus_system_time_secs": 2.19,
      "cpus_user_time_secs": 7.74,
      "mem_anon_bytes": 103370752,
      "mem_file_bytes": 86016,
      "mem_limit_bytes": 557842432,
      "mem_mapped_file_bytes": 45056,
      "mem_rss_bytes": 103456768,
      "timestamp": 1422535725.82344
    }
 }
```

you will get following metric:

```json
"frameworks.marathon-0_7_6.executors.com_domain_group_anotherApp.mem_mapped_file_bytes": 45056
```
If there will be nesting you will end up with nested version of marathon in framework branch. But I think `executor_id`  could be spited so there will be tree `com/domain/group/`. Good catch I will fix it.